mzLib/MzLibUtil/ClassExtensions.cs

```diff
@@ -19,12 +19,96 @@
+
+
+            // If there is a missed cleavage, then there will be a label on K and a Label on X modification.
+            // It'll be like [label]|[label] which complicates the positional stuff a little bit. Therefore, 
+            // RemoveSpecialCharacters will remove the "|", to ease things later on. 
+            RemoveSpecialCharacters(ref fullSeq);
+            MatchCollection matches = regex.Matches(fullSeq);
+            int captureLengthSum = 0;
+            foreach (Match match in matches)
+            {
+                GroupCollection group = match.Groups;
+                string val = group[1].Value;
+                int startIndex = group[0].Index;
+                int captureLength = group[0].Length;
+
+                List<string> modList = new List<string>();
+                modList.Add(val);
+
+                // The position of the amino acids is tracked by the positionToAddToDict variable. It takes the 
+                // startIndex of the modification Match and removes the cumulative length of the modifications
+                // found (including the brackets). The difference will be the number of nonmodification characters, 
+                // or the number of amino acids prior to the startIndex in the sequence. 
+                int positionToAddToDict = startIndex - captureLengthSum;
+
+                // Handle N terminus indexing
+                if ((positionToAddToDict == 0) && !IncludeNTerminus)
+                {
+                    positionToAddToDict++;
+                }
+
+                // Handle C terminus indexing
+                if ((fullSeq.Length == startIndex + captureLength) && IncludeCTerminus)
+                {
+                    positionToAddToDict++;
+                }
+
+                // check to see if key already exist
+                // if the already key exists, update the current position with the capture length + 1.
+                // otherwise, add the modification to the dict.
+                if (modDict.ContainsKey(positionToAddToDict))
+                {
+                    modDict[positionToAddToDict].Add(val);
+                }
+                else
+                {
+                    modDict.Add(positionToAddToDict, modList);
+                }
+                captureLengthSum += captureLength;
+            }
+            return modDict;
+        }
+
+        /// <summary>
+        /// Fixes an issue where the | appears and throws off the numbering if there are multiple mods on a single amino acid.
+        /// </summary>
+        /// <param name="fullSeq"></param>
+        /// <param name="replacement"></param>
+        /// <param name="specialCharacter"></param>
+        /// <returns></returns>
+        public static void RemoveSpecialCharacters(ref string fullSeq, string replacement = @"", string specialCharacter = @"\|")
+        {
+            // next regex is used in the event that multiple modifications are on a missed cleavage Lysine (K)
```

Comment:
mods with transition metals will be problematic. suggest:
                          sequence = sequence.Replace("[I]", "");
                            sequence = sequence.Replace("[II]", "");
                            sequence = sequence.Replace("[III]", "");

Reply:
Updated the bracket match logic to ignore a closing bracket if it matches the pattern for a metal ion modification, namely `"\[I+\]"`. New mod search pattern is  `@"\[(.+?)\](?<!\[I+\])"`